### ISSUE TITLE
Validators not applied to parameters with BindingSource = Form

### ISSUE THREAD

**Issue by steverb38** [SharpGrip/FluentValidation.AutoValidation#7]
Hi,

Thanks for providing this package for auto FluentValidation!

My first attempt as using this was trying to apply a Validator to a Form parameter but I see the code specifically only applies validators to Body or Query parameters. Not being a web developer particularly, is there a specific reason for this restriction?

More a question in education (for me) perhaps?

Steve.

**Comment by steverb38** [comment]
Hi,

My use-case was submitting a form for a file upload to the server, along with some associated parameters to control the behaviour of the API in processing the uploaded file. The API action method looks something like:

```C#
[HttpPost]
public async Task<ActionResult<MyResponse>> Post([FromForm] MyRequest request)
{
...
}
```

where MyRequest is like:

```C#
public class MyRequest
{
    public IFormFile File { get; set; }

    public string Param1 { get; set; }

    public bool Param2 { get; set; }

    ....
}
```

There were then some validations to perform on the Param1, Param2,... values to ensure together they made sense.

Steve.

**Comment by steverb38** [comment]
OK - I updated a local copy of FluentValidationAutoValidationActionFilter.cs on line 55 to allow a BindingSource of Form as well as Body and Query.

![image](https://github.com/SharpGrip/FluentValidation.AutoValidation/assets/7479033/8eaefb51-a9e1-4116-bc5b-3a830f7e858a)

That seemed to then work OK in my test scenario.  No other testing done though!

**Comment by mvdgun** [comment]
@steverb38 that's great to hear! I actually have been playing with the idea to expose the binding sources via the options builder and via the data annotations attributes to make it all configurable. The default would be `BindingSource.Body`, `BindingSource.Body` and `BindingSource.Query`. The only issue I see with the defaults is if I should consider this a breaking change.

**Comment by mvdgun** [comment]
Opened a PR for it: https://github.com/SharpGrip/FluentValidation.AutoValidation/pull/8/files. Would appreciate it if you could take a look at it :).

**Comment by steverb38** [comment]
Hi. Fantastic! I'll take a look later today. 👍

**Comment by mvdgun** [comment]
Hi @steverb38, I just released v1.2.0 which includes the configurable binding sources.
